app/feature/feature-forever/src/main/kotlin/com/hedvig/android/feature/forever/ForeverViewModel.kt

```diff
@@ -20,19 +22,19 @@ import giraffe.ReferralsQuery
 import giraffe.fragment.ReferralFragment
 
 internal class ForeverViewModel(
-  private val foreverRepository: ForeverRepository,
-  private val getReferralTermsUseCase: GetReferralsInformationUseCase,
+  foreverRepositoryProvider: Provider<ForeverRepository>,
```

Comment:
Great that you can inject the Provider interface instead!

Reply:
Yeah, this way the tests are able to just do `Provider { fakeImpl() }` as we discussed here https://hedviginsurance.slack.com/archives/D02DE2ZGLS1/p1695369859355719?thread_ts=1695367706.676759&cid=D02DE2ZGLS1
So the tests need not know anything about demo mode whatsoever.